modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+
+[discrete]
+== `k8sDeleteResource`
+
+It deletes resources from the cluster, based on the provided model, resource. The garbage collection works based on 'Foreground'|'Background', can be configured with propagationPolicy property in provided model or passed in json. It returns a promise that resolves to the response of kind Status. In case of failure promise gets rejected with HTTP error response.
+
+....
+{ kind: 'DeleteOptions', apiVersion: 'v1', propagationPolicy }
+....
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |which are passed as key-value pair in the map.
+
+|`` |options.model - k8s model
+
+|`` |options.resource - The resource to be deleted.
+
+|`` |options.path - Appends as subpath if provided
+
+|`` |options.queryParams - The query parameters to be included in the
+URL.
+
+|`` |options.requestInit - The fetch init object to use. This can have
+request headers, method, redirect, etc. See more \{@link
+https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
+}
+
+|`` |options.json - Can control garbage collection of resources
+explicitly if provided else will default to model's "propagationPolicy".
+|===
+
+[discrete]
+== `k8sListResource`
+
+Lists the resources as an array in the cluster, based on provided options. It returns a promise that resolves to the response.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |Which are passed as key-value pairs in the map
+
+|`` |options.model - k8s model
+
+|`` |options.queryParams - The query parameters to be included in the
+URL and can pass label selector's as well with key "labelSelector".
+
+|`` |options.requestInit - The fetch init object to use. This can have
+request headers, method, redirect, etc. See more \{@link
+https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
+}
+|===
+
+[discrete]
+== `k8sListResourceItems`
+
+Same interface as \{@link k8sListResource} but returns the sub items. It returns the apiVersion for the model i.e `group/version`.
+
+[discrete]
+== `getAPIVersionForModel`
+
+Provides apiVersion for a k8s model.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`model` |k8s model
+|===
+
+[discrete]
+== `getGroupVersionKindForResource`
+
+Provides a group, version, and kind for a resource. It returns the group, version, kind for the provided resource. If the resource does not have an API group, group "core" will be returned.If the resource has an invalid apiVersion then it'll throw Error.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resource` |k8s resource
+|===
+
+[discrete]
+== `getGroupVersionKindForModel`
+
+Provides a group, version, and kind for a k8s model. This returns the group, version, kind for the provided model. If the model does not have an apiGroup, group "core" will be returned.
```

Comment:
s/k8s/Kubernetes

~~This returns the group, version, kind for the provided model.~~ Redundant?

s/group "core" will be/the group core is